src/components/cc-env-var-form/cc-env-var-form.types.d.ts

```diff
@@ -1 +1,20 @@
-type ContextType = "env-var" | "env-var-simple" | "env-var-addon" | "exposed-config" | "config-provider";
+import { EnvVarMode, Variable } from "../common.types";
+
+export type EnvVarFormState = EnvVarFormStateLoading | EnvVarFormStateLoaded | EnvVarFormStateError;
+
+interface EnvVarFormStateLoading {
+  type: 'loading';
+}
+
+interface EnvVarFormStateLoaded {
+  type: 'loaded';
+  variables: Array<Variable>;
+  mode: EnvVarMode;
+  saving: boolean;
```

Comment:
question: I expected to have a `'saving'` state instead of this boolean on the `loaded` state, I imagine you considered it, I'm curious to understand why you preferred this?

In the smart part, when saving is done, this would made us going explicitly back to `loaded`, instead of setting `saving` to `false` and relying that we should already be in state `loaded`.

Reply:
There is no good reason to do like I did.
So lets go for a proper `'saving'` state type.